lib/check_graphite/projection.rb

```diff
@@ -0,0 +1,39 @@
+require 'attime'
+require 'bigdecimal'
+require 'nagios_check'
+require 'linear-regression'
+
+module CheckGraphite
+  module Projection
+    def self.included(base)
+      base.on '--projection FUTURE_TIMEFRAME', :default => '2days' do |timeframe|
+        options.send(:processor=, method(:projected_value))
+        options.send(:timeframe=, timeframe)
+      end
+    end
+
+    def projected_value(datapoints)
+      ys, xs = datapoints.transpose
+      lr = Regression::Linear.new(xs, ys)
+      future = CheckGraphite.attime(options.timeframe, xs[-1])
+      value = lr.predict(future.to_i)
+      p = Regression::CorrelationCoefficient.new(xs, ys).pearson
+      store_value options.name, value
+      # Unfortunately, nagios_check converts both primary and
+      # secondary values to float. Hence manual assignment instead:
+      @values['p-value'] = format_float(p.abs)
+      store_message "#{options.name}=#{format_float(value)} in #{options.timeframe}"
+      return value
+    end
+
+    private
+
+     def format_float(v)
+       if v.nan?
+         'undefined'
+       else
+         BigDecimal.new(v, 3).to_s('F')
```

Comment:
sprintf() can only do number of decimals, but here we deal with numbers that can be any float, so there is no way to know how many decimals are relevant, if any. Thus we instead do number of significant digits. Why 3? It works for our use case; it will need to be configurable eventually, but I think it would be a rare use case that cares about changes < 1/1000. I don't think the helper merits being named a test unit in its own right. However, this means that the "integration" tests that tests against the full output should check that the number is formatted correctly. Fixed.